### DISCUSSION TITLE
We're up to Preview 5 for v11

### DISCUSSION THREAD

**Discussion by donnyv** [AvaloniaUI/Avalonia#10456] [General]
I've been waiting for v11 to have an official release, so that I can start building my project. But we're up too preview 5. Is there any timeline to when the official v11 will be released?

**Comment by timunie** [comment]
As often in Open Source software, it's done when it's done. But to give at least a rough estimation, preview 5 is probably the last preview. It was planned to have a first RC as the next preview release.

**Reply by robloo** [reply]
I think a preview 6 might be a good idea. Preview 5 had so many issues it couldn't be tested as much. If using a 3rd party library it's also difficult to test with latest master due to breaking changes.

An RC means we will release if all is ok. I doubt preview 5 has been tested well enough yet though personally. If the nugets are automatically generated and a release isn't that much extra work, I would be in favor of another quick preview that can be better tested without all the blockers.

**Reply by ahopper** [reply]
I think RC should mean no features or tweaks will be added just bug fixes, if the team feel they are at that point then calling the next release an RC might encourage more testing and tempt 3rd party libs to be updated.

**Reply by donnyv** [reply]
> I think RC should mean no features or tweaks will be added just bug fixes, if the team feel they are at that point then calling the next release an RC might encourage more testing and tempt 3rd party libs to be updated.

Yeah thats my biggest issue right now. So many of the third party libraries are behind on compatibility with the v11 previews. I feel like they're waiting for the RC. Plus I really want to start with an API thats not going to change much.

**Reply by mysteryx93** [reply]
I agree that preview5 has many issues that make it hard to test. The latest FluentAvalonia preview5.1 targets the nightly build. I tried switching to the nightly build to be able to publish/trim my assembly, but got too many issues including Avalonia Behaviors not supporting it yet so I switched back.

**Reply by robloo** [reply]
I'm now working with latest nightly and FluentAvalonia 5.1. There are still several issues. It's the most unstable I've seen Avalonia before and there are lots of unexpected crashes that did not occur before.

I'm doubling down that we need another preview before RC or this is going to be less than ideal publicity.

I'm going to start filing general issues. Don't have time at the moment to narrow down everything I'm seeing.

**Reply by mysteryx93** [reply]
Looking at the current state of Avalonia, I understand why it took so long before preview5. I'm also humbled by the scale of this project -- making Avalonia11 production-ready for Android, iOS and Web is an absolutely massive undertaking. I can only imagine the scale of rewriting under the hood that they had to go through to make it work.

**Reply by mysteryx93** [reply]
To make things clear. Publishing and trimming is not possible in preview5.

FluentAvalonia-preview5.1 requires `11.0.999-cibuild0030643-beta`, while Avalonia Behaviors nightly build doesn't support after `11.0.999-cibuild0030320-beta`. How am I supposed to test that?

**Reply by grokys** [reply]
I agree that we need a `preview6` before an RC. I think though if we're going to do #9944 for 11.0 then we need to get that finished first, as IMO a change as big as that will need a preview phase.

**Reply by mysteryx93** [reply]
One more note. There are many breaking changes between each preview, making it hard to have all 3rd party components compatible with the same Avalonia build. And yes I understand the reason, they want to do all "future" breaking changes before v11 goes out.

**Comment by MikeCodesDotNET** [comment]
To add some context to this discussion. The most significant reason for the pace of v11 moving to stable is our aversion to breaking APIs between major versions. That is, we want to avoid making significant changes to the APIs until v12, giving you a stable API to build against. 

V11 has a number of very substantial changes that obviously take time to create. An example of this is our compositional renderer, which has been a multi-year project! 

While our community is growing and we receive some incredible contributions, a small team does most of the work on Avalonia. We'd love to grow this team, which is a big reason for creating Avalonia XPF. If you'd like to support Avalonia UI, then sponsorship does go a long way in helping us.

**Reply by ahopper** [reply]
Just a thought, I wonder if giving a couple of days warning before a preview or RC release might encourage wider testing by Avalonia's keener users and improve the experience of previews for others.

**Reply by mysteryx93** [reply]
> Just a thought, I wonder if giving a couple of days warning before a preview or RC release might encourage wider testing by Avalonia's keener users and improve the experience of previews for others.

I don't think so. Most users spend their time in their code and know of a release via NuGet upgrades. They come here to have more information about a release, or when they have issues.

Considering the major improvements of Avalonia11, I think that a lot of people are testing it.

**Reply by robloo** [reply]
I agree some discussion and prior notification about RC release date is valuable. It will help scheduling the open issues as well.

**Reply by grokys** [reply]
@robloo @tomenscape if we can get #9944 finished and get a preview 6 out this week, then I think we're in a position where we can start to think about a date for the RC (though it's possible that there are other destabilizing changes I've forgotten).

**Reply by mysteryx93** [reply]
There are serious issues with CompiledBindings that may be because of Cecile; if it requires changing the underlying engine, that can be a pretty big change.

**Reply by maxkatz6** [reply]
As far as I can tell, there is one serious bug with compiled bindings (https://github.com/AvaloniaUI/Avalonia/issues/10485) that might block whole idea of enabling them by default (and also make trimming harder).

I still yet need to dig into https://github.com/AvaloniaUI/Avalonia/issues/10494, and understand why it works on windows as reported. Looks like something that worked before.

**Reply by TomEdwardsEnscape** [reply]
@grokys, closing #9944 this week is not possible. There are still many large and more challenging controls to convert, not least `TextBox` and `ItemsControl`, which don't have PRs open yet.

Completing the current PR (#10370) is feasible though.

**Reply by robloo** [reply]
TextBox isn't going to be done -- only TextBlock is planned.

**Comment by mysteryx93** [comment]
Any idea when Preview6 will be released? I wanted to publish some assemblies with Trimming, and that won't happen now. It requires using nightly builds, and I tried getting master builds for Avalonia, Avalonia Behaviors and FluentAvalonia running -- way to many problems show up everywhere that distract me from actually getting things done.

Better wait for Preview6. I hope it won't take as long as for Preview5.
